### DISCUSSION TITLE
OBS studio with WHIP Client and my Server

### DISCUSSION THREAD

**Discussion by DrakF12** [obsproject/obs-studio#9429] [Development Questions]
I am testing my Server using WHIP as the signal. But I am confused……
When I use OBS studio like this 
<img width="714" alt="截屏2023-08-14 20 51 09" src="https://github.com/obsproject/obs-studio/assets/43041736/bb6c8cd6-85b5-425a-a340-8002892fbf9e">
I can check the OBS studio send the SDP offer in Wireshark:
<img width="905" alt="截屏2023-08-14 20 54 59" src="https://github.com/obsproject/obs-studio/assets/43041736/7967f6f4-570c-4302-bb0a-97e7223768e7">
And the SDP answer from my server is:
<img width="896" alt="截屏2023-08-14 20 57 50" src="https://github.com/obsproject/obs-studio/assets/43041736/0b23f684-d225-43c3-8b6c-2845a45d8c48">

I think the OBS studio should send the STUN package(binding request), but it did not.
The OBS send the "FIN" to shut down the TCP connection.

The output log of the OBS studio like this:

20:32:13.579: [obs-webrtc] [whip_output: 'simple_stream'] PeerConnection state is now: Connecting
20:32:13.676: ==== Streaming Start ===============================================
20:32:30.882: [obs-webrtc] [whip_output: 'simple_stream'] PeerConnection state is now: Closed
20:32:30.977: [obs-webrtc] [whip_output: 'simple_stream'] DELETE request for resource URL failed. HTTP Code: 405
20:32:30.977: Output 'simple_stream': stopping
20:32:30.977: Output 'simple_stream': Total frames output: 492
20:32:30.977: Output 'simple_stream': Total drawn frames: 522
20:32:30.981: ==== Streaming Stop ================================================

I do not understand why not establish the ICE. 
Are there some details of SDP ignored？  I need your help.

**Comment by Sean-Der** [comment]
Hey @DrakF12!

I don't believe `libdatachannel` supports a remote that is `ice-lite`. When grepping through the code base I don't see anything.

Is this server public? Am I able to test against it? I will update `libdatachannel` to support a remote that is `ice-lite`. It will be a very simple change.

thanks for testing OBS + WHIP out!

**Comment by DrakF12** [comment]
Thanks for your reply. This server is not public，but I will ask my leader for permission.
        I am developing signalling server  based on WHIP for WebRtc. Therefore，I have used OBS Studio to test my WHIP server.I will try to update my ICE-options to ICE2.Now，I still have one question about debugging OBS studio. Can we offer some scripts(python or Lua) to recode the detail SDP offer/answer, STUN package(binding request/response) without building the OBS studio? Or debug mode for user? The log of OBS studio is too simple.I just need more details about the interactive process not build the whole project. I am not good at building OBS. There are so many building problems, which made me confused.  
    
        If you can offer some suggestions, I will be very happy.
